timex/timex.go

```diff
@@ -0,0 +1,57 @@
+// Package timex should be called Rolex.
+package timex
+
+import (
+	"context"
+	"time"
+)
+
+// Ticker ...
+type Ticker struct {
+	C <-chan time.Time
+
+	tt   *time.Ticker
+	done chan bool
+	c    chan time.Time
+}
+
+// NewTicker ...
+func NewTicker(d time.Duration) *Ticker {
+	c := make(chan time.Time, 1)
+	ticker := Ticker{
+		C:    c,
+		tt:   time.NewTicker(d),
+		done: make(chan bool),
+		c:    c,
+	}
+
+	go func() {
+		for {
+			select {
+			case <-ticker.done:
+				return
+			case t := <-ticker.tt.C:
+				ticker.c <- t
+			}
+		}
+	}()
+
+	return &ticker
+}
+
+// Stop ...
+func (t *Ticker) Stop() {
+	t.tt.Stop()
```

Comment:
If you move this `Stop` call to the `select case <-ticker.done:`, you could get rid of the `tt` field and instead make it a local variable inside the goroutine above.

Reply:
ok, i think i got it all. i could then also remove the Stop() in the goroutines done clause (I think).

Reply:
I would still call Stop() because I'm not sure what happens under the hood with garbage collection on an unstopped timer. There's some funny stuff with the runtime package with timers.